master/buildbot/test/unit/test_process_build.py

```diff
@@ -572,6 +573,45 @@ def testStepDoneHaltOnFailureFlunkOnWarnings(self):
         self.assertEqual(terminate, False)
         self.assertEqual(b.result, FAILURE)
 
+    def testStepDonePauseAgentOnFailure(self):
+        b = self.build
+        b.results = [SUCCESS]
+        b.result = SUCCESS
+        b.remote = Mock()
+        b.slavebuilder = Mock()
+        step = FakeBuildStep()
+        step.pauseAgentOnFailure = True
+        terminate = b.stepDone(FAILURE, step)
+        self.assertEqual(terminate, False)
+        self.assertEqual(b.result, FAILURE)
+        b.slavebuilder.slave.slave_status.setPaused.assert_called_once_with(True)
+
+    def testStepDonePauseAgentOnException(self):
+        b = self.build
+        b.results = [SUCCESS]
+        b.result = SUCCESS
+        b.remote = Mock()
+        b.slavebuilder = Mock()
+        step = FakeBuildStep()
+        step.pauseAgentOnFailure = True
```

Comment:
Could you add a test where paused agent on failure is false, Btw in buildbot the concept is slave and not agent maybe we should rename to be consistent with buildbot?